### PR TITLE
MM-30431 Detox/E2E: Added e2e for MM-T3196, MM-T3210, MM-T3211, MM-T3262, MM-T3264

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added tests for MM-T3196, MM-T3210, MM-T3211, MM-T3262, MM-T3264 in
    - `detox/e2e/test/smoke_test/about.e2e.js`
    - `detox/e2e/test/smoke_test/advanced_settings.e2e.js`
    - `detox/e2e/test/smoke_test/channel_add_members.e2e.js`
    - `detox/e2e/test/smoke_test/direct_messages.e2e.js`
- Added new screens
    - `detox/e2e/support/ui/screen/about.js`
    - `detox/e2e/support/ui/screen/advanced_settings.js`
    - `detox/e2e/support/ui/screen/user_profile.js`
- Added testIDs

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-30431

TM4J Cases:
- [MM-T3196](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3196)
- [MM-T3210](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3210)
- [MM-T3211](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3211)
- [MM-T3262](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3262)
- [MM-T3264](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T3264)

Screen Shot 2021-04-12 at 6 53 17 PM
Screen Shot 2021-04-12 at 6 52 52 PM
Screen Shot 2021-04-12 at 6 54 02 PM
Screen Shot 2021-04-12 at 6 53 07 PM

### CODE REVIEW

File: app/components/sidebars/main/channels_list/channel_item/channel_item.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default class ChannelItem extends PureComponent {
         channelId: PropTypes.string.isRequired,
         channel: PropTypes.object,
         currentChannelId: PropTypes.string.isRequired,
-        displayName: PropTypes.string.isRequired,
+        displayName: PropTypes.string,
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/channel_item/channel_item.js:25]: Why not required anymore?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -651,10 +651,10 @@ EXTERNAL SOURCES:
 SPEC CHECKSUMS:
   boost-for-react-native: 39c7adb57c4e60d6c5479dd8623128eb5b3f0f2c
   BVLinearGradient: e3aad03778a456d77928f594a649e96995f1c872
-  DoubleConversion: cde416483dac037923206447da6e1454df403714
+  DoubleConversion: cf9b38bf0b2d048436d9a82ad2abe1404f11e7de
```
Comment (enahum) [inline ios/Podfile.lock:654]: ????

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/components/sidebars/main/channels_list/channel_item/channel_item.js (hunk 1 above)
Comment (josephbaylon) [reply app/components/sidebars/main/channels_list/channel_item/channel_item.js:25]: @enahum It was causing an error when cache is deleted. Also on line 93, there's a check for 
```
        if (!displayName) {
            return null;
        }
```

which to me didn't look like it's required if it's checking that.

File: ios/Podfile.lock (hunk 2 above)
Comment (josephbaylon) [reply ios/Podfile.lock:654]: I got this diff from my local `npm install`. Not sure if it was missed by previous PRs but I don't think there should be difference with my local on pod file if I'm using the same version of pods. I can revert this if you think this shouldn't be the case. Lmk.

File: app/components/sidebars/main/channels_list/channel_item/channel_item.js (hunk 1 above)
Comment (enahum) [reply app/components/sidebars/main/channels_list/channel_item/channel_item.js:25]: nice catch!

**Review by enahum — APPROVED** [review]

**Review by saturninoabril — APPROVED** [review]
Awesome, thanks Joseph! Tested and passed. LGTM.
Screen Shot 2021-04-14 at 12 15 55 AM
Screen Shot 2021-04-14 at 12 17 01 AM
Screen Shot 2021-04-14 at 12 20 19 AM
Screen Shot 2021-04-14 at 12 22 38 AM

**Pull request merged**